flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -247,6 +256,18 @@ public int getOperatorsMemoryInMB() {
 	 */
 	public boolean isMatching(ResourceProfile required) {
 
+		if (this == ANY) {
+			return true;
+		}
+
+		if (this == required) {
+			return true;
+		}
+
+		if (this == UNKNOWN) {
```

Comment:
Should UNKNOWN match UNKNOWN resources?

Reply:
Yes. That case is covered by the second if-branch (`this == required`).